src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/System.Text/Rune.cs

```diff
@@ -0,0 +1,1436 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
i kept this as close to the same as i could.  Areas i changed:

1. made the type internal.
2. used normal `throw new ...` instead of `ThrowHelper`.
3. Remove CLSCompliant attributes
4. #ifdef'ed out `GlobalizationMode.Invariant` sections.  They're optimizations that don't ever apply to us and caused me to need to pull in too much extra stuff.